pynumdiff/utils/evaluate.py

```diff
@@ -86,18 +86,16 @@ def robust_rme(u, v, padding=0, M=6):
     :param np.array[float] v: e.g. estimated smoothed signal, reconstructed from derivative
     :param int padding: number of snapshots on either side of the array to ignore when calculating
         the metric. If :code:`'auto'`, defaults to 2.5% of the size of inputs
-    :param float M: Huber loss parameter in units of ~1.4*mean absolute deviation, intended to approximate
-        standard deviation robustly.
+    :param float M: Huber loss parameter in units of ~1.4*mean absolute deviation of population of residual
+        errors, intended to approximate standard deviation robustly.
 
     :return: (float) -- Robust root mean error between u and v
     """
     if padding == 'auto': padding = max(1, int(0.025*len(u)))
     s = slice(padding, len(u)-padding) # slice out data we want to measure
-    N = s.stop - s.start
 
     sigma = stats.median_abs_deviation(u[s] - v[s], scale='normal') # M is in units of this robust scatter metric
-    if sigma < 1e-6: sigma = 1 # guard against divide by zero
-    return np.sqrt(2*np.mean(utility.huber((u[s] - v[s])/sigma, M))) * sigma
+    return np.sqrt(2*np.mean(utility.huber(u[s] - v[s], M*sigma)))
```

Comment:
I realized dividing the first arg by $\sigma$ and scaling the whole Huber function by $\sigma^2$ (such that a single $\sigma$ ends up outside the $\sqrt{\cdot}$) is equivalent to scaling *up* $M$ by $\sigma$, and the latter is mildly easier to write down and interpret, so I've updated the paper with it as well.